test/WorldTests/FieldComputationTest.cpp

```diff
@@ -99,13 +99,11 @@ TEST(FieldComputationTest, point_in_polygon) {
     EXPECT_FALSE(FieldComputations::pointIsInField(testField, testPoint, -1.0));
 
     // Pick the corner cases (boundary value analysis) for our defence area in the pointInDefenceArea function
-    testPoint = testField.getLeftPenaltyLineTop() + Vector2(0.3, 0.3);
-
-    // TODO: FIX THIS
-    // EXPECT_TRUE(FieldComputations::pointIsInDefenseArea(testField, testPoint, true, 0.3, 0.2));
+    testPoint = testField.getLeftPenaltyLineTop() + Vector2(0.3 - smallChanges, 0.3 - smallChanges);
+    EXPECT_TRUE(FieldComputations::pointIsInDefenseArea(testField, testPoint, true, 0.3, 0.2));
```

Comment:
```suggestion
    // Check if a point very close but still inside the defense area is correctly evaluated as being in the defense area
    testPoint = testField.getLeftPenaltyLineTop() + Vector2(0.3 - smallChanges, 0.3 - smallChanges);
    EXPECT_TRUE(FieldComputations::pointIsInDefenseArea(testField, testPoint, true, 0.3, 0.2));
```
This way the reader doesn't have to do the mental math and decide if the smallchanges should be added or subtracted, and it's super clear what exactly you are testing, so if the tests break people can fix them more easily

Reply:
Adding 1 line of comment per 2 lines of code is way too much. As currently explained in the code comments this is a boundary value analysis test case. This is way enough documentation. Also not everything has a reason behind it: 0.2 and 0.3 where just small enough values which is the reason why I used them. But anything different could also have been used. Moreover testing is only important once the specification is clear and I expect these specification to be changed in the near future. Thus spending too much time on test cases is a waste of time, because test cases have to be changed once the specification is changed...

Reply:
People need to understand what you are testing, and if that takes 2 comments or 0, I don't care. I hope you can see that my suggestion makes it instantly clear, without even needing to look at the code, what is being tested. 

I doubt this specification will be changed, as we are playing robot soccer, and you are checking if the defense area boundaries work well. Unless I missed a drastic rule change, we will have a defense area for quite a while.

Reply:
I think it would really help if you look at the documentation I wrote in the code review. This leaves nothing to the imagination